spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskSchedulerController.java

```diff
@@ -77,16 +78,10 @@ public TaskSchedulerController(SchedulerService schedulerService) {
 	 */
 	@RequestMapping(value = "", method = RequestMethod.GET)
 	@ResponseStatus(HttpStatus.OK)
-	public PagedResources<ScheduleInfoResource> list(
+	public PagedResources<ScheduleInfoResource> list(Pageable pageable,
 			PagedResourcesAssembler<ScheduleInfo> assembler) {
 		List<ScheduleInfo> result = this.schedulerService.list();
-		int resultSize = result.size();
-		Pageable pageable = PageRequest.of(0,
-				(resultSize == 0) ? resultSize = 1 : resultSize); //handle empty result set
-
-		Page<ScheduleInfo> page = new PageImpl<>(result, pageable,
-				result.size());
-		return assembler.toResource(page, taskAssembler);
+		return assembler.toResource(new PageImpl<>(result, pageable, result.size()), taskAssembler);
```

Comment:
why do we still need a `new PageImpl<>` here?

Reply:
We still need to do this because the `scheduler` service [doesn't have the `list(Pageable)` implemented](https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultSchedulerService.java#L170) yet.